src/data/option/basic.lean

```diff
@@ -106,6 +106,18 @@ by cases x; simp
   x.map f = some b ↔ ∃ a, x = some a ∧ f a = b :=
 by cases x; simp
 
+lemma map_eq_none {α β} {x : option α} {f : α → β} :
+  f <$> x = none ↔ x = none :=
+by { cases x; simp only [map_none, map_some, eq_self_iff_true] }
+
+@[simp] lemma map_eq_none' {x : option α} {f : α → β} :
+  x.map f = none ↔ x = none :=
+by { cases x; simp only [map_none', map_some', eq_self_iff_true] }
+
+lemma map_congr {f g : α → β} {x : option α} (h : ∀ a ∈ x, f a = g a) :
+  option.map f x = option.map g x :=
```

Comment:
I think this lemma would be better if you left it without the `x` in the result (and maybe also the hypothesis. If you do that, then you can claim that `map` is injective

Reply:
How would you phrase the hypothesis then? I've found this lemma useful for the cases when `x` is some complex expression, and `f a = g a` for all `some a` for that complex `x`, but not in the general case.

There's already an `option.map_injective' lemma.